bolt-modules/dir/lib/puppet/functions/dir/children.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Returns an array containing all of the filenames except for "." and ".." in the given directory.
+Puppet::Functions.create_function(:'dir::children', Puppet::Functions::InternalFunction) do
+  # @param dirname Absolute path or Puppet module name.
+  # @return Array of files in the given directory.
+  # @example List filenames from an absolute path.
+  #   dir::children('/home/user/subdir/')
+  # @example List filenames from a Puppet file path.
+  #   dir::children('puppet_agent')
+  dispatch :children do
+    scope_param
+    required_param 'String', :dirname
+    return_type 'Array'
+  end
+
+  def children(scope, dirname)
+    # Send Analytics Report
+    Puppet.lookup(:bolt_executor) {}&.report_function_call(self.class.name)
+    puppet_mod = scope.compiler.environment.module(dirname)&.path
+
+    # Sort for testability
+    Dir.children(puppet_mod || dirname).sort
```

Comment:
While not strictly necessary, I do think it would be nice to rescue the Ruby error here and raise a Bolt error with a cleaner message. Currently if the directory doesn't exist the message will be similar to: `No such file or directory @ dir_initialize - foo"`

Reply:
Hm, I think if we do that we'll lose the line and file we get from Pal, and that doesn't seem worth recalculating:
```
lucy ~/githubs/bolt (GH-1981) $ be bolt plan run bolt -t localhost
Starting: plan bolt
Finished: plan bolt in 0.0 sec
{
  "kind": "dir/children",
  "msg": "Could not list files in fake: No such file or directory @ dir_initialize - /home/lucy/githubs/bolt/fake",
  "details": {
  }
}
lucy ~/githubs/bolt (GH-1981) $ be bolt plan run bolt -t localhost
Starting: plan bolt
Finished: plan bolt in 0.0 sec
{
  "kind": "bolt/pal-error",
  "msg": "No such file or directory @ dir_initialize - /home/lucy/githubs/bolt/fake",
  "details": {
    "file": "/home/lucy/githubs/bolt/plans/init.pp",
    "line": 4,
    "column": 10
  }
}
```

I can modify the error message and reraise it, but honestly "Could not list files in fake: No such file or directory @ dir_initialize - /home/lucy/githubs/bolt/fake" doesn't seem any better than just "No such file or directory @ dir_initialize - /home/lucy/githubs/bolt/fake"